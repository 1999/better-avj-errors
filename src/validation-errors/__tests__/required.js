import { getFixturePath } from 'jest-fixtures';
import { readFileSync } from 'fs';

import RequiredValidationError from '../required';

async function getSchemaAndData(name) {
  const schemaPath = await getFixturePath(__dirname, name, 'schema.json');
  const schema = JSON.parse(readFileSync(schemaPath, 'utf8'));
  const dataPath = await getFixturePath(__dirname, name, 'data.json');
  const data = JSON.parse(readFileSync(dataPath, 'utf8'));
  return [schema, data];
}

describe('Required', () => {
  it('prints correctly for missing required prop', async () => {
    const [schema, data] = await getSchemaAndData('required');

    const error = new RequiredValidationError({
      keyword: 'required',
      dataPath: '',
      schemaPath: '#/required',
      params: { missingProperty: 'id' },
      message: `should have required property 'id'`,
    });

    expect(error.print(schema, data)).toMatchSnapshot();
  });
});
