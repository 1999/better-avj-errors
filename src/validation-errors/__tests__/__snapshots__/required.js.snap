// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Required prints correctly for missing required prop 1`] = `
Array [
  "[31m[1mREQUIRED[22m[39m[31m should have required property 'id'[39m
",
  "[32m- Extra[39m
[31m+ Missing[39m

[32m- Object {}[39m
[31m+ Object {[39m
[31m+   \\"id\\": \\"required\\",[39m
[31m+ }[39m",
]
`;
